File: QUEUE/01.c
//QUEUE IMPLEMENTATION USING ARRAY
#include<stdio.h>
#define N 5
int arr[N];
int front = -1;
int rear = -1;
void enqueue(){
    int x;
    printf("enter the value to be inserted:\n");
    scanf("%d",&x);
    if(rear==N-1){
        printf("Queue if Full!");
    }else if(rear==-1 && front==-1){
        front = rear = 0;
        arr[rear]=x;
    }else{
        rear++;
        arr[rear] = x;
    }
     printf("%d enqued\n",x);
}
void dequeu(){
    if(rear==-1 && front==-1){
        printf("empty!");
    }else if(rear == front){
        printf("The removed element is %d\n", arr[front]);
        front = rear = -1;
    }else{
        printf("the removed element is %d\n",arr[front]);
        front++;
    }
}
void display(){
    int i;
    if(rear==-1 && front==-1){
        printf("empty!");
    }
    else{
        for(i=front;i<rear+1;i++){
            printf("%d ",arr[i]);
        }
    }
}
int main(){
    int n;
    while(1){
    printf("\nEnter your choice:\n");
    scanf("%d",&n);
    switch(n){
        case 1:
        enqueue();
        break;
        case 2:
        dequeu();
        break;
        case 3:
        display();
        break;
        case 4:
        return 0;
        break;
        case 5:
        printf("enter valid input");
        break;
    }
    }

}

File: QUEUE/01.py
# IMPLEMENTATION OF QUEUE IN PYTHON
class queue:
    def __init__(self,size):
        self.queue = []
        self.size = size

    def enqueue(self, item):
        if len(self.queue) == self.size:
            print("Queue is full")
        else:
            self.queue.append(item)
            print(f"Enqueued: {item}")


    def dequeue(self):
        if self.is_empty():
            print("Queue is empty")
        else:
            removed = self.queue.pop(0)
            print(f"Dequeued: {removed}")

    
    def peek(self):
        if self.is_empty():
            print("Queue is empty")
        else:
            print(f"Front element: {self.queue[0]}")

    
    def display(self):
        if self.is_empty():
            print("Queue is empty")
        else:
            print("Queue elements:", self.queue)

  
    def is_empty(self):
        return len(self.queue) == 0




q = queue(5)
q.enqueue(10)
q.enqueue(20)
q.enqueue(30)
q.display()
q.dequeue()
q.display()
q.peek()
